mypy/meet.py

```diff
@@ -486,13 +486,15 @@ def visit_type_var(self, t: TypeVarType) -> ProperType:
     def visit_instance(self, t: Instance) -> ProperType:
         if isinstance(self.s, Instance):
             si = self.s
-            if t.type == si.type:
+            if t.type == si.type and len(t.args) == len(si.args):
```

Comment:
Btw why do we need the `and ...` here if we anyway use `zip()` below?

Reply:
Oh, no, we definitely do not. The `and` was my first approach for fixing this, before I decided I liked the zip version more. (This check does work, though, so that could be an approach?)